packages/plugin-json/src/lib/amplience-schema-transformers.ts

```diff
@@ -158,10 +158,32 @@ export const objectProperties = (
             other: (type) =>
               ampliencePropertyType(prop, type, schema, schemaHost),
           }),
+          ...extensionProperty(prop, schema),
```

Comment:
I think we should not add the extension property here. Instead, in the ampliencePropertyType function you can extend this block for your logic.
```
 if (isObjectType(node) && node.astNode) {
      if (hasDirective(prop, "amplienceLink")) {
        return contentLink(node.astNode, schemaHost);
      }
      if (hasDirective(prop, "amplienceReference")) {
        return contentReference(node.astNode, schemaHost);
      }

      if (hasDirective(node.astNode, "amplienceContentType")) {
        return inlineContentReference(node.astNode, schemaHost);
      }

      return inlineObject(node.astNode, schema, schemaHost);
    }
```

Reply:
thanks for suggestions, definitely looks cleaner.